packages/terra-form-select/src/shared/_MenuUtil.js

```diff
@@ -195,6 +195,9 @@ class MenuUtil {
   static findNext(object, value) {
     const options = MenuUtil.flatten(object, true);
     const index = options.findIndex(({ props }) => MenuUtil.isEqual(props.value, value));
+    if (options.length - 1 === index) {
```

Comment:
Should this be
`if (index + 1 >== options.length - 1)`

It seems like you are trying to wrap the selection.

Reply:
If we add this check focus is skipping last item in dropdown while navigate using arrow key